inst/tests/tests.Rraw

```diff
@@ -21041,3 +21041,37 @@ test(2304.100, set(copy(DT), i=2L, j=c("L1", "L2"), value=list(list(NULL), list(
 
 # the integer overflow in #6729 is only noticeable with UBSan
 test(2305, { fread(testDir("issue_6729.txt.bz2")); TRUE })
+
+# Tests for print.data.table handling index columns, list classes, and unknown classes in #6806
+
+# Test for covering classes[col_name] <- "index"
+DT <- data.table(A = 1:3, B = 4:6)
+setindex(DT, A)
+test(2306.1, {print(DT); capture.output(print(DT))}, 
+     c("   A B",
+       "1: 1 4",
+       "2: 2 5",
+       "3: 3 6"))
+# Test for covering if (is.list(cls)) cls <- unlist(cls)
+DT <- data.table(A = 1:3, B = list(NULL, NULL, NULL))
+test(2306.2, {print(DT); capture.output(print(DT))}, 
+     c("   A      B",
+       "1: 1 [NULL]",
+       "2: 2 [NULL]",
+       "3: 3 [NULL]"))
+# Test for covering if (length(cls) == 0) cls <- "unknown"
+DT <- data.table(A = 1:3, B = I(list()))
+test(2306.3, {print(DT); capture.output(print(DT))}, 
+     c("   A      B",
+       "1: 1 [NULL]",
+       "2: 2 [NULL]",
+       "3: 3 [NULL]"))
+# Test for covering classes[col_name] <- "unknown"
+DT <- data.table(A = 1:3, B = 4:6, C = 7:9)
+if ("D" %in% colnames(DT)) DT[, D := NULL]
+test(2306.4, {print(DT); capture.output(print(DT))}, 
+     c("   A B C",
+       "1: 1 4 7",
+       "2: 2 5 8",
+       "3: 3 6 9"))
```

Comment:
In order to exercise the index-printing code, make sure to create an index using `setindex(...)` and use `print(..., datatable.show.indices=TRUE)` or `test(..., output = ..., options = c(datatable.show.indices=TRUE))`. In order to reproduce the warning, make sure to create more than one column, otherwise `rbind()` silently recycles the length-1 vector `abbs`.